eng/scripts/live-test-resource-cleanup.ps1

```diff
@@ -86,6 +85,16 @@ foreach ($rg in $toDelete)
   }
 }
 
+# Get purgeable resources already in a deleted state coerced into a collection even if empty.
+$purgeableResources = Get-PurgeableResources
+$allPurgeCount = $purgeableResources.Count
+
+# Filter down to the ones that we can actually perge.
+$purgeableResources = $purgeableResources.Where({ $purgeDate = $_.ScheduledPurgeDate -as [DateTime]; (!$purgeDate -or $now -gt $purgeDate) })
```

Comment:
This is now how you filter them. You need to check EnableSoftDelete. You've also not purged the vaults early enough. The default is 90d, so you're effectively leaving hundreds of vaults un-purged.

Reply:
There's no point in purging vaults in the future. Once the purge date is hit, they are purged automatically.